### PR TITLE
Miscellaneous code cleanup

### PR DESCRIPTION
**Pull request by StefanSpieker**
Minor performance improvements and do not overload a method if only super is called

### Proposed changelog entries

* N/A

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [ ] (If applicable) Jira issue is well described
- [ ] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [ ] Appropriate autotests or explanation to why this change has no tests
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

**Review by timja — APPROVED** [review]

File: core/src/main/java/hudson/tasks/CommandInterpreter.java
Comment (daniel-beck) [inline core/src/main/java/hudson/tasks/CommandInterpreter.java:130]: Does this change behavior in case of `null` values?

File: core/src/main/java/hudson/Functions.java
Comment (daniel-beck) [inline core/src/main/java/hudson/Functions.java:1515]: Not sure this is worth the reduced readability?

(Also, can't `javac` do this by now?)

Comment (StefanSpieker) [reply core/src/main/java/hudson/Functions.java:1515]: I think `javac` will do something automatically if we are >= JDK9. The change might still help with Java 8

**Comment by basil** [comment]
@StefanSpieker This looks almost ready to merge to me, but I am a little concerned by the code Daniel highlighted as well ("Does this change behavior in case of null values?"). Would you be willing to just revert that hunk and leave the rest of this diff as is? Then I would be happy to mark this PR as `ready-for-merge` and we can continue with the refactoring to `CommandInterpreter` in some future PR.

**Comment by basil** [comment]
I reverted about half of the changes in this PR, as I either couldn't understand them after thinking about it for 5 minutes, or because I felt that they were a net neutral rather than a net positive. What remains are things that I feel are objectively safe:

- The change to `UpdateCenter` is fine; see [Arrays of Wisdom of the Ancients](https://shipilev.net/blog/2016/arrays-wisdom-ancients/).
- The change to `UpdateSite` is fine because it avoids unnecessary boxing.
- The change to `UnwrapSecurityExceptionFilter` is fine because [JLS § 15.20.2](https://docs.oracle.com/javase/specs/jls/se11/html/jls-15.html#jls-15.20.2) guarantees `instanceof` does a null check.

These three changes are ready to merge from my perspective. If you want to proceed with the other changes, that is fine, but please open a new PR and explain the reasoning behind each one. If the reasoning is sound I am happy to approve the others. I really appreciate these cleanup PRs, but unfortunately providing a bunch of risky changes without reasoning and not replying to feedback for months is not going to work.

**Review by res0nance — APPROVED** [review]

**Review by basil — APPROVED** [review]
This PR is now ready for merge. We will merge it after approximately 24 hours if there is no negative feedback. Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process. Thanks!

**Pull request merged**
